src/client/lixa_tx.c

```diff
@@ -53,6 +53,70 @@
 #endif /* LIXA_TRACE_MODULE */
 #define LIXA_TRACE_MODULE   LIXA_TRACE_MOD_CLIENT_TX
 
+static char *tx_open_profile = NULL;
+static GMutex profile_mutex = G_STATIC_MUTEX_INIT;
+
+int lixa_tx_set_profile(int *txrc, const char *profile)
+{
+    enum Exception {
+        G_STRDUP_ERROR,
+        NONE
+    } excp;
+    int ret_cod = LIXA_RC_INTERNAL_ERROR;
+    *txrc = TX_FAIL;
+
+    LIXA_TRACE_INIT;
+    LIXA_CRASH_INIT;
+    LIXA_TRACE(("lixa_tx_set_profile\n"));
+
+    g_mutex_lock( &profile_mutex );
+    TRY {
+        if (tx_open_profile) {
+            free(tx_open_profile);
+            tx_open_profile = NULL;
+        }
+        if (profile) {
+	    tx_open_profile = g_strdup(profile);
+            if (tx_open_profile == NULL) {
+                THROW( G_STRDUP_ERROR );
+            } else {
+                THROW( NONE );
+            }
+        } else {
+            THROW( NONE );
+        }
+    } CATCH {
+        switch (excp) {
+        case NONE:
+            ret_cod = LIXA_RC_OK;
+            *txrc = TX_OK;
+            break;
+        case G_STRDUP_ERROR:
```

Comment:
This still requires a little rework to avoid issues during problem determination:

    case G_STRDUP_ERROR:
      ret_cod = LIXA_RC_G_STRDUP_ERROR;
      *txrc = TX_FAIL;
      break;
    default:
       ret_cod = LIXA_RC_INTERNAL_ERROR; 
       *txrc = TX_FAIL;
       break;

Reply:
Done.